lib/elixir/test/elixir/code_test.exs

```diff
@@ -126,9 +126,9 @@ defmodule CodeTest do
 
     test "raises when string_to_quoted!/2 is used and no atom is found with :existing_atoms_only" do
       unknown_atom = ":there_is_no_such_atom"
-
-      assert catch_error(Code.string_to_quoted!(unknown_atom, existing_atoms_only: true)) ==
-               :badarg
+      assert_raise SyntaxError, fn ->
```

Comment:
Can we assert on the error message? It is the second argument. Then we can verify that it indeed returns file and line and includes the atom name, which are the reasons for this change!

Reply:
Done!